ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/ShelleyHFC.hs

```diff
@@ -198,16 +215,126 @@ instance ShelleyBasedEra era => HasPartialLedgerConfig (ShelleyBlock era) where
+    return $ completeLedgerConfig
+      (Proxy @(ShelleyBlock era))
+      (fixedEpochInfo epochSize slotLength)
+      partialConfig
+
+  encodeNodeToClient ccfg version ledgerConfig = mconcat [
+      encodeListLen 3
+    , encodeNodeToClient
+        @_
+        @(ShelleyPartialLedgerConfig era)
+        ccfg
+        version
+        (toPartialLedgerConfig (Proxy @(ShelleyBlock era)) ledgerConfig)
+    , toCBOR @EpochSize epochSize
+    , encode @SlotLength slotLength
+    ]
+    where
+      unwrap = either
+        (error "ShelleyLedgerConfig contains a non-fixed EpochInfo")
+        id
+      epochInfo_ = epochInfoWithErr (shelleyLedgerGlobals ledgerConfig)
+      epochSize = unwrap $ epochInfoSize epochInfo_ 0
+      RelativeTime t1 = unwrap $ epochInfoSlotToRelativeTime epochInfo_ 1
+      slotLength = mkSlotLength t1
+
+-- | This instance uses the invariant that the EpochInfo in a
+-- ShelleyPartialLedgerConfig is always just a dummy value.
+instance ShelleyBasedEra era
+      => SerialiseNodeToClient (ShelleyBlock era) (ShelleyPartialLedgerConfig era) where
+  decodeNodeToClient ccfg version = do
+    enforceSize "ShelleyPartialLedgerConfig era" 13
+    ShelleyPartialLedgerConfig
+      <$> ( ShelleyLedgerConfig
+        <$> fromCBOR @(CompactGenesis era)
+        <*> (SL.Globals
+              (toPureEpochInfo dummyEpochInfo)
+              <$> fromCBOR @Word64
+              <*> fromCBOR @Word64
+              <*> fromCBOR @Word64
+              <*> fromCBOR @Word64
+              <*> fromCBOR @Word64
+              <*> fromCBOR @Word64
+              <*> fromCBOR @Natural
+              <*> fromCBOR @Word64
+              <*> fromCBOR @ActiveSlotCoeff
+              <*> fromCBOR @SL.Network
+              <*> fromCBOR @SystemStart
+            )
+        <*> fromCBOR @(SL.TranslationContext era)
+      )
+      <*> decodeNodeToClient @(ShelleyBlock era) @TriggerHardFork  ccfg version
+
+  encodeNodeToClient ccfg version
+    (ShelleyPartialLedgerConfig
+      (ShelleyLedgerConfig
+        myCompactGenesis
+        (SL.Globals
+          _epochInfo
+          slotsPerKESPeriod'
+          stabilityWindow'
+          randomnessStabilisationWindow'
+          securityParameter'
+          maxKESEvo'
+          quorum'
+          maxMajorPV'
+          maxLovelaceSupply'
+          activeSlotCoeff'
+          networkId'
+          systemStart'
+        )
+        translationContext
+      )
+      triggerHardFork
+    )
+      = encodeListLen 15
+        <> toCBOR @(CompactGenesis era) myCompactGenesis
+        <> toCBOR @Word64 slotsPerKESPeriod'
+        <> toCBOR @Word64 stabilityWindow'
+        <> toCBOR @Word64 randomnessStabilisationWindow'
+        <> toCBOR @Word64 securityParameter'
+        <> toCBOR @Word64 maxKESEvo'
+        <> toCBOR @Word64 quorum'
+        <> toCBOR @Natural maxMajorPV'
+        <> toCBOR @Word64 maxLovelaceSupply'
+
+        -- TODO shouldn't all these things use `SerialiseNodeToClient` instances?
```

Comment:
Reifying this TODO as a PR comment.

Reply:
These are all things defined in lower level packages. I don't see a benefit to making explicit `SerialiseNodeToClient` for these types, so I'll leave it like this for now and remove the comment.